src/openrct2/ride/TrackDesign.cpp

```diff
@@ -1251,9 +1252,8 @@ static int32_t track_design_place_maze(rct_track_td6* td6, int16_t x, int16_t y,
 
                     if (_trackDesignPlaceOperation == PTD_OPERATION_1)
                     {
-                        cost = game_do_command(
-                            mapCoord.x, 0 | rotation << 8, mapCoord.y, ((z / 16) & 0xFF) | (1 << 8),
-                            GAME_COMMAND_PLACE_RIDE_ENTRANCE_OR_EXIT, -1, 0);
+                        auto res = RideEntranceExitPlaceAction::TrackPlaceQuery({ mapCoord.x, mapCoord.y, z }, true);
```

Comment:
How come this requires a static function rather than calling QueryNested?

Reply:
The game command was really odd in that it called a separate part of the query code and never called the execute section of that code. It didn't make sense for it to be a game command. You will notice it takes slightly different parameters and validates them differently. I'm not sure what the best approach would be.

Reply:
Hm yeah, I can only think of separating the path in Query for when the ride id is -1 (0xFF) but then its quite the annoyance to pass temporary parameters for everything else.